test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs

```diff
@@ -154,26 +154,30 @@ internal void Create(TestAsset targetTestAsset, string testProjectsSourceFolder,
 
             var projectXml = XDocument.Load(sourceProject);
 
+            if (projectXml.Root is null)
+            {
+                throw new InvalidOperationException($"The project file '{sourceProject}' does not have a root element.");
+            }
             var ns = projectXml.Root.Name.Namespace;
 
-            if (ProjectSdk != null)
+            if (projectXml.Root.Attribute("Sdk") is not null && ProjectSdk != null)
             {
-                projectXml.Root.Attribute("Sdk").Value = ProjectSdk;
+                projectXml.Root.Attribute("Sdk")!.Value = ProjectSdk;
             }
```

Comment:
Might be better to do something like this. Not sure it will work exactly like this since I'm not sure of the types used here. Might need to play around with it.

```suggestion
            var sdkAttribute = projectXml.Root.Attribute("Sdk");
            if (sdkAttribute is not null && ProjectSdk != null)
            {
                sdkAttribute.Value = ProjectSdk;
            }
```